apps/prairielearn/src/ee/lib/ai-grading/ai-grading-util.ts

```diff
@@ -160,14 +161,94 @@ export async function generatePrompt({
   }
 
   // Student response
-  messages.push({
-    role: 'user',
-    content: `The student submitted the following response: \n<response>\n${submission_text} \n<response>\nHow would you grade this? Please return the JSON object.`,
-  });
+  messages.push(
+    generateSubmissionMessage({
+      submission_text,
+      submitted_answer,
+    }),
+  );
 
   return { messages };
 }
 
+/**
+ * Parses the student's submission text and answer to generate a message for the AI model.
+ */
+function generateSubmissionMessage({
+  submission_text,
+  submitted_answer,
+}: {
+  submission_text: string;
+  submitted_answer: Record<string, any> | null;
+}): ChatCompletionMessageParam {
+  const message_content: ChatCompletionContentPart[] = [];
+
+  message_content.push({
+    type: 'text',
+    text: 'The student submitted the following response: \n<response>\n',
+  });
+
+  const $submission_html = cheerio.load(submission_text);
+  let submissionTextSegment = '';
+
+  $submission_html
```

Comment:
nit: A high level comment explaining that we walk through the HTML nodes, from "top" to "bottom" to construct a AI-readable version of the submission.